pkg/database/realm.go

```diff
@@ -358,28 +360,17 @@ func (r *Realm) BeforeSave(tx *gorm.DB) error {
 		r.AddError("longCodeDuration", "must be no more than 24 hours")
 	}
 
-	if r.EnableENExpress {
-		if !strings.Contains(r.SMSTextTemplate, SMSENExpressLink) {
-			r.AddError("SMSTextTemplate", fmt.Sprintf("must contain %q", SMSENExpressLink))
-		}
-		if strings.Contains(r.SMSTextTemplate, SMSRegion) {
-			r.AddError("SMSTextTemplate", fmt.Sprintf("cannot contain %q - this is automatically included in %q", SMSRegion, SMSENExpressLink))
-		}
-		if strings.Contains(r.SMSTextTemplate, SMSLongCode) {
-			r.AddError("SMSTextTemplate", fmt.Sprintf("cannot contain %q - the long code is automatically included in %q", SMSLongCode, SMSENExpressLink))
-		}
-
-	} else {
-		// Check that we have exactly one of [code] or [longcode] as template substitutions.
-		if c, lc := strings.Contains(r.SMSTextTemplate, SMSCode), strings.Contains(r.SMSTextTemplate, SMSLongCode); !(c || lc) || (c && lc) {
-			r.AddError("SMSTextTemplate", fmt.Sprintf("must contain exactly one of %q or %q", SMSCode, SMSLongCode))
+	r.validateSMSTemplate(r.SMSTextTemplate)
```

Comment:
This API is a bit weird since it doesn't return anything. I don't have a better solution though.